Description: Upstream changes introduced in version 0.1-2~wip
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 libsml (0.1-2~wip) unstable; urgency=low
 .
   * Added information for pkg-config
   * Eased dependency for libuuid from version 2.17 to 2.16
     as there were no changes in the symbols
   * Changed architecture for developement files to 'all'
 .
 The person named in the Author field signed this changelog entry.
Author: Steffen Vogel <info@steffenvogel.de>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- /dev/null
+++ libsml-0.1/sml.pc
@@ -0,0 +1,13 @@
+prefix=/usr 
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: libSML
+Description: Library for the Smart Messaging Language (SML)
+Version: 0.1
+URL: http://github.com/dailab/libsml
+Requires: uuid >= 2.16
+Libs: -L${libdir} -lsml
+Libs.private: -luuid
+Cflags: -I${includedir}/sml
--- libsml-0.1.orig/test/Makefile
+++ libsml-0.1/test/Makefile
@@ -1,10 +1,11 @@
 UNAME := $(shell uname)
 CFLAGS += -I../sml/include/ -Wall
+LIBSML = ../sml/lib/libsml.a
+LIBS = -luuid
+
 ifeq ($(UNAME), Linux)
 LIBS = -luuid
 endif
-LIBSML = ../sml/lib/libsml.a
-
 
 UNITY = \
 	unity/unity.o \
@@ -30,22 +31,18 @@ test_run: libsml test
 	@./test
 
 test : $(UNITY) $(OBJS) $(LIBSML)
-	$(CC) $(CFLAGS) $(LIBS) $(LIBSML) $^ test_main.c -o test
+	$(CC) $(CFLAGS) $(LIBS) $^ test_main.c -o test
 
 .PHONY: code
 libsml : 
 	@$(MAKE) -C ../sml
 
 %.o : %.c
-	$(CC) $(CFLAGS) $(LIBS) -c $^ -o $@
+	$(CC) $(CFLAGS) -c $^ -o $@
 
 .PHONY: clean
 clean :
 	@rm -f unity/*.o
 	@rm -f src/*.o
-
-
-
-
-
+	@rm -f test
 
--- libsml-0.1.orig/sml/Makefile
+++ libsml-0.1/sml/Makefile
@@ -4,6 +4,10 @@ CFLAGS += -I./include/ -fPIC -g -Wall
 # Available Flags:
 # _NO_UUID_LIB - compile without uuid lib
 
+ifeq ($(UNAME), Linux)
+LIBS = -luuid
+endif
+
 LIB_DIR=./lib
 OBJ_LIB=$(LIB_DIR)/libsml.o
 DYN_LIB=$(LIB_DIR)/libsml.so
@@ -45,7 +49,7 @@ libsml: $(ST_LIB) $(OBJ_LIB)
 endif
 
 $(DYN_LIB): $(OBJS)
-	$(LD) -shared -o $@ $^
+	$(LD) $(LIBS) -shared -o $@ $^
 
 $(OBJ_LIB): $(OBJS)
 	$(LD) -r -o $@ $^
--- libsml-0.1.orig/examples/Makefile
+++ libsml-0.1/examples/Makefile
@@ -1,17 +1,17 @@
 UNAME := $(shell uname)
 CFLAGS += -I../sml/include/ -g -Wall
 OBJS = sml_server.o
+LIBSML = ../sml/lib/libsml.a
 
 ifeq ($(UNAME), Linux)
-LIBS = -L../sml/lib/ ../sml/lib/libsml.o -luuid
-endif
-ifeq ($(UNAME), Darwin)
-LIBS = -L../sml/lib/ ../sml/lib/libsml.o
+LIBS = -luuid
 endif
 
+sml_server : $(OBJS) $(LIBSML)
+	$(CC) $(CFLAGS) $(OBJS) $(LIBS) $(LIBSML) -o sml_server
 
-sml_server : $(OBJS)
-	$(CC) $(CFLAGS) $(LIBS) -o sml_server $(OBJS)
+%.o : %.c
+	$(CC) $(CFLAGS) -c $^ -o $@
 
 .PHONY: clean
 clean:
